src/models/encryption-wallet.ts

```diff
@@ -1,43 +1,43 @@
 import { ethers } from "ethers";
+import { ViewStateFacet } from '@sarcophagus-org/sarcophagus-v2-contracts'
 
+// TODO -- update to more appropriate derivation path that isn't BIP44
 const PATH_WITHOUT_INDEX = "m/44'/60'/0'/0/"
 
 export class EncryptionWallet {
   public wallet: ethers.utils.HDNode;
-  public index: number;
+  private viewStateFacet: ViewStateFacet;
 
-  constructor(encryptionHdWallet: ethers.utils.HDNode) {
+  constructor(
+    encryptionHdWallet: ethers.utils.HDNode,
+    viewStateFacet: ViewStateFacet
+    ) {
     this.wallet = encryptionHdWallet;
-    const index1000 = this.getHdNodeAtIndex(1000);
-    this.derivePrivateKeyFromPublicKey(index1000.publicKey);
+    this.viewStateFacet = viewStateFacet;
   }
 
-  derivePrivateKeyFromPublicKey(publicKey: string, index: number = 0): string {
+  deriveHdWalletFromPublicKey(publicKey: string, index: number = 0): ethers.utils.HDNode {
```

Comment:
On second look, "derive" in the function name is a bit of a misnomer. Had me thinking this was trying to *derive* a whole wallet from just a public key. 

A simpler "get" would be less confusing. Maybe something along the lines of `getHdWalletOfPublicKey`?